exonum/src/blockchain/transaction.rs

```diff
@@ -808,23 +811,41 @@ mod tests {
         panic::catch_unwind(panic::AssertUnwindSafe(|| panic!(val))).unwrap_err()
     }
 
-    fn create_blockchain() -> (Blockchain, BTreeMap<Hash, Box<Transaction>>) {
+    fn create_blockchain() -> Blockchain {
         let service_keypair = crypto::gen_keypair();
         let api_channel = mpsc::channel(1);
-        (
-            Blockchain::new(
-                MemoryDB::new(),
-                Vec::new(),
-                service_keypair.0,
-                service_keypair.1,
-                ApiSender::new(api_channel.0),
-            ),
-            BTreeMap::new(),
+        Blockchain::new(
+            MemoryDB::new(),
+            vec![Box::new(TxResultService) as Box<Service>],
+            service_keypair.0,
+            service_keypair.1,
+            ApiSender::new(api_channel.0),
         )
+
+    }
+
+    struct TxResultService;
+    impl Service for TxResultService {
+        fn service_id(&self) -> u16 {
+            1
+        }
+
+        fn service_name(&self) -> &'static str {
+            "test service"
+        }
+
+
+        fn state_hash(&self, _: &Snapshot) -> Vec<Hash> {
+            vec![]
+        }
+
+        fn tx_from_raw(&self, raw: RawTransaction) -> Result<Box<Transaction>, ::encoding::Error> {
+            Ok(Box::new(TxResult::from_raw(raw)?))
+        }
     }
 
     transactions! {
-        Transactions {
+        TestTx {
```

Comment:
We commonly use `tx` as singular form, so it should be `TestTxs`, but I prefer to name it `TestTransactions`.